### ISSUE TITLE
Sequencer: Implement Mempool

### ISSUE THREAD

**Issue by eyusufatik** [chainwayxyz/citrea#82]
First task originating from #22 

Having a good mempool is a must.

A mempool should be able to identify invalid transactions according to state + state after the same account's transactions in the mempool.

So a mempool should be able to order transactions by their fee paid, by nonce of the same account etc.

This is not an easy task so we won't write a mempool from scratch.

We are going to use reth's Pool. From my understanding `Pool::eth_pool()` with NoopBlobStore would work for us. [See here](https://github.com/paradigmxyz/reth/blob/cbf6324594a20e3a99e58abcdff3819f52067940/crates/transaction-pool/src/lib.rs#L271).

The sequencer will call `pool.pooled_transactions()` to receive all valid transactions and will put them in a block. (So change on the actual sequencer is minimal.) Right now we don't have many transactions on our mempool so we can include all in a block no problem. We'll have another task for it.

Another change on the sequencer is `eth_sendRawTransaction`, call `pool.add_transaction()` to put transactions in the mempool.

Bare minimum tests to write:

- [x] Transaction with nonce lower then account's nonce on state should not be accepted by mempool.
- [x] Transaction with nonce higher then account's nonce should be accepted by the mempool but shouldn't be received by the sequencer (so it doesn't end up in the block)
- [ ] Transaction with not enough balance (to pay for the transaction) should be accepted by the mempool but shouldn't be received by the sequencer. Only after another transaction that funds that account lands in the mempool it should be received by the sequencer. 
- [x] Two transactions from two different accounts, send the one with lower fee first and the higher fee second. The transaction with higher fee should land first.
- [ ] Set up a contract with a function that does `coinbase.transfer(1 ether)`. Call this contract with a transaction with gas price 0. The transaction should land in a block.
- [x] Send a transaction that pays less base fee then required. Publish block, tx should not be in block but should still be in the mempool.
- [ ] Transaction replacement by higher gas fee. Note that there is a parameter for increase that is checked, test under and above of this parameter.

I'll update tests as I come up with different ideas.
